flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -468,6 +423,59 @@ private static Request createRequest(
+
+        final SSLHandlerFactory sslHandlerFactory = restConfiguration.getSslHandlerFactory();
+        ChannelInitializer<SocketChannel> initializer =
+                new ChannelInitializer<SocketChannel>() {
+                    @Override
+                    protected void initChannel(SocketChannel socketChannel) {
+                        try {
+                            // SSL should be the first handler in the pipeline
+                            if (sslHandlerFactory != null) {
+                                socketChannel
+                                        .pipeline()
+                                        .addLast(
+                                                "ssl",
+                                                sslHandlerFactory.createNettySSLHandler(
+                                                        socketChannel.alloc(),
+                                                        targetAddress,
+                                                        targetPort));
+                            }
+
+                            socketChannel
+                                    .pipeline()
+                                    .addLast(new HttpClientCodec())
+                                    .addLast(
+                                            new HttpObjectAggregator(
+                                                    restConfiguration.getMaxContentLength()));
+
+                            for (OutboundChannelHandlerFactory factory :
+                                    outboundChannelHandlerFactories) {
+                                Optional<ChannelHandler> channelHandler =
+                                        factory.createHandler(configuration);
+                                if (channelHandler.isPresent()) {
+                                    socketChannel.pipeline().addLast(channelHandler.get());
+                                }
+                            }
+
+                            socketChannel
+                                    .pipeline()
+                                    .addLast(new ChunkedWriteHandler()) // required for
+                                    // multipart-requests
+                                    .addLast(
+                                            new IdleStateHandler(
+                                                    restConfiguration.getIdlenessTimeout(),
+                                                    restConfiguration.getIdlenessTimeout(),
+                                                    restConfiguration.getIdlenessTimeout(),
+                                                    TimeUnit.MILLISECONDS))
+                                    .addLast(new ClientHandler());
+                        } catch (Throwable t) {
+                            t.printStackTrace();
+                            ExceptionUtils.rethrow(t);
+                        }
+                    }
+                };
+        bootstrap.handler(initializer);
```

Comment:
Is this function not called from multiple threads? Now we introduce race here...